pallets/protos/src/lib.rs

```diff
@@ -134,11 +137,10 @@ pub mod pallet {
 		StorageMap<_, Identity, Hash256, Proto<T::AccountId, T::BlockNumber>>;
 
 	#[pallet::storage]
-	pub type ProtosByTag<T: Config> = StorageMap<_, Blake2_128Concat, Tags, Vec<Hash256>>;
+	pub type ProtosByTag<T: Config> = StorageDoubleMap<_, Blake2_128Concat, Tags, Blake2_128Concat, Hash256, bool>;
 
 	#[pallet::storage]
-	pub type ProtosByOwner<T: Config> =
-		StorageMap<_, Blake2_128Concat, ProtoOwner<T::AccountId>, Vec<Hash256>>;
+	pub type ProtosByOwner<T: Config> = StorageDoubleMap<_, Blake2_128Concat, ProtoOwner<T::AccountId>, Blake2_128Concat, Hash256, bool>;
```

Comment:
like the above

Reply:
Are all those types Iterable by default? You didn't need to change type? Just a trait right?

Reply:
By "types", are you referring to the Storage Items (StorageValue, StorageMap, StorageDoubelMap and StorageNMap) or to `ProtoOwner<T::AccountId>`, `Hash256` and `bool`? @sinkingsugar